### DISCUSSION TITLE
Voxelgrid from voxel + transformation ?

### DISCUSSION THREAD

**Discussion by StephaneKazmierczak** [isl-org/Open3D#3902] [Q&A]
Hello, I am looking to apply transformations to a voxel from a numpy array. 

Ex : 
```python
array([[[0, 0, 0],
        [0, 0, 0],
        [1, 0, 0]],

       [[0, 0, 0],
        [0, 0, 0],
        [0, 0, 0]],

       [[0, 0, 0],
        [0, 0, 0],
        [0, 1, 1]]])
```

Is there an object that would take the above representation and then allow for transformation (rotation/flip). I have looked at the documentation but it doesn;t seems the voxelgrid works like that.

**Comment by ssheorey** [comment]
Hi @StephaneKazmierczak are you looking to transform the entire voxel grid? What does the NumPy array represent?

```py
from open3d.geometry import VoxelGrid, PointCloud
pcd = PointCloud()
pcd.points = o3d.utility.Vector3dVector(xyz)   # xyz is a (N,3) numpy array
vg = VoxelGrid.create_from_point_cloud(pcd, voxel_size)
vg.transform(transform_matrix_4x4)
```

**Reply by saurabheights** [reply]
@ssheorey This wont work. Translation and rotation are not **yet** defined for Voxelgrid (See https://github.com/isl-org/Open3D/blob/master/cpp/open3d/geometry/VoxelGrid.cpp#L98).

**Reply by ssheorey** [reply]
Hey nice catch! Yes, this is something we would need to VoxelGrid.
